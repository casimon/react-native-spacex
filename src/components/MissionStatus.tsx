import React from 'react';
import {
  Text,
  View,
  StyleSheet,
  StyleProp,
  TextStyle,
  ViewStyle
} from 'react-native';

import colors from '../constants/colors';

const styles = StyleSheet.create({
  missionStatus: {
    width: 100,
    marginHorizontal: 6,
    paddingVertical: 3,
    paddingHorizontal: 6,
    backgroundColor: '#ccc',
    borderRadius: 2,
    marginVertical: 6
  },
  missionStatusText: {
    fontSize: 12,
    color: colors.white,
    textAlign: 'center'
  },
  success: {
    backgroundColor: '#359A53'
  },
  failure: {
    backgroundColor: '#9A3635'
  },
  tentative: {
    backgroundColor: '#D3BF47'
  }
});

interface Props {
  success?: boolean;
  tentative?: boolean;
  style?: StyleProp<ViewStyle>;
  styleText?: StyleProp<TextStyle>;
}

class MissionStatus extends React.PureComponent<Props> {
  render() {
    const {success, tentative} = this.props;

    return (
      <View
        style={[
          styles.missionStatus,
          this.props.style,
          success
            ? styles.success
            : tentative
            ? styles.tentative
            : styles.failure
        ]}>
        <Text style={[styles.missionStatusText, this.props.styleText]}>
          {success ? 'Successful' : tentative ? 'Tentative' : 'Failure'}
        </Text>
      </View>
    );
  }
}

export default MissionStatus;
