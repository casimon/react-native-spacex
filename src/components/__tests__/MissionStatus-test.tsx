/**
 * @format
 */

import 'react-native';
import React from 'react';
import MissionStatus from '../MissionStatus';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders mission status with all props undefined', () => {
  const tree = renderer.create(<MissionStatus />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders mission status with success false', () => {
  const tree = renderer.create(<MissionStatus success={false} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders mission status with success true', () => {
  const tree = renderer.create(<MissionStatus success={false} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders mission status with tentative false', () => {
  const tree = renderer.create(<MissionStatus tentative={false} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders mission status with tentative true', () => {
  const tree = renderer.create(<MissionStatus tentative={false} />).toJSON();
  expect(tree).toMatchSnapshot();
});
