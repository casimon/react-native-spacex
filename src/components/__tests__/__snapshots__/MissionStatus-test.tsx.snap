// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders mission status with all props undefined 1`] = `null`;

exports[`renders mission status with success false 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ccc",
        "borderRadius": 2,
        "marginHorizontal": 6,
        "marginVertical": 6,
        "paddingHorizontal": 6,
        "paddingVertical": 3,
        "width": 100,
      },
      undefined,
      Object {
        "backgroundColor": "#9A3635",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 12,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    Failure
  </Text>
</View>
`;

exports[`renders mission status with success true 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ccc",
        "borderRadius": 2,
        "marginHorizontal": 6,
        "marginVertical": 6,
        "paddingHorizontal": 6,
        "paddingVertical": 3,
        "width": 100,
      },
      undefined,
      Object {
        "backgroundColor": "#9A3635",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 12,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    Failure
  </Text>
</View>
`;

exports[`renders mission status with tentative false 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ccc",
        "borderRadius": 2,
        "marginHorizontal": 6,
        "marginVertical": 6,
        "paddingHorizontal": 6,
        "paddingVertical": 3,
        "width": 100,
      },
      undefined,
      Object {
        "backgroundColor": "#9A3635",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 12,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    Failure
  </Text>
</View>
`;

exports[`renders mission status with tentative true 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ccc",
        "borderRadius": 2,
        "marginHorizontal": 6,
        "marginVertical": 6,
        "paddingHorizontal": 6,
        "paddingVertical": 3,
        "width": 100,
      },
      undefined,
      Object {
        "backgroundColor": "#9A3635",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 12,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    Failure
  </Text>
</View>
`;
